src/assets.ts

```diff
@@ -576,31 +579,39 @@ async function addLaunchJsonIfNecessary(generator: AssetGenerator, operations: A
             return resolve();
         }
 
-        // NOTE: We will NOT attempt to merge in the existing launch.json configurations
-        // because in the startup prompt case, we will not attempt to create a launch.json if it
-        // already exists, and in the command case, we delete the launch.json file, but the VS
-        // Code API will return old configurations anyway, which we do NOT want.
-
         const programLaunchType = generator.computeProgramLaunchType();
         const launchJsonConfigurations: string = generator.createLaunchJsonConfigurations(programLaunchType);
-        const configurationsMassaged: string = indentJsonString(launchJsonConfigurations);
 
-        const launchJsonText = `
-{
-   // Use IntelliSense to find out which attributes exist for C# debugging
-   // Use hover for the description of the existing attributes
-   // For further information visit https://github.com/OmniSharp/omnisharp-vscode/blob/master/debugger-launchjson.md
-   "version": "0.2.0",
-   "configurations": ${configurationsMassaged}
-}`;
 
-        fs.writeFile(generator.launchJsonPath, launchJsonText.trim(), err => {
-            if (err) {
-                return reject(err);
-            }
+        if (!fs.pathExistsSync(generator.launchJsonPath)) {
+            // when launch.json does not exist, create it and write all the content directly
+            const configurationsMassaged: string = indentJsonString(launchJsonConfigurations);
+            const launchJsonText = `
+            {
+                // Use IntelliSense to find out which attributes exist for C# debugging
+                // Use hover for the description of the existing attributes
+                // For further information visit https://github.com/OmniSharp/omnisharp-vscode/blob/master/debugger-launchjson.md
+                "version": "0.2.0",
+                "configurations": ${configurationsMassaged}
+            }`;
+            
+            fs.writeFile(generator.launchJsonPath, launchJsonText.trim(), err => {
+                if (err) {
+                    return reject(err);
+                }
 
-            resolve();
-        });
+                resolve();
+            });
+        } 
+        else {
+            // when launch.json exists append our configurations
```

Comment:
>// when launch.json exists append our configurations [](start = 12, length = 52)

I feel like we shouldn't go through the VS Code APIs, as that isn't going to work with our comments.

Reply:
What if user has comments in their file? Then we would lose their comments.

Reply:
Okay updating would lose all comments, ours and theirs. I need a different way.

Reply:
If you have ideas how to do that please share, I saw that jsonc-parser has quite a lot of capabilities, so maybe it could be done via that, but if there is a known easy way I am all ears.

Reply:
Looks like jsonc-parser has utility functions named `modify` and `applyEdits`. You would use `modify` to generate an array of Edits that you could then pass to `applyEdits` in order to get an updated json string to save.

Reply:
Thanks. I reached out to the maintainer of that package, so I will wait till he gets back to me, maybe there is some example that I can follow.

Reply:
With the current approach we don't lose ours nor the user comments. (Unless they place their comments into the tasks that we overwrite, but they will be prompted to confirm the overwrite).